src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -470,6 +511,30 @@ export abstract class InstructorSessionBasePageComponent {
     modal.componentInstance.requestId = resp.error.requestId;
     modal.componentInstance.errorMessage = resp.error.message;
   }
+
+  triggerModelChange(data: SessionEditFormModel): void {
+    const { submissionStartDate, submissionEndDate, submissionStartTime, submissionEndTime } = data;
+
+    const startDate = new Date(submissionStartDate.year, submissionStartDate.month, submissionStartDate.day);
+    const endDate = new Date(submissionEndDate.year, submissionEndDate.month, submissionEndDate.day);
+
+    if (startDate > endDate) {
+      this.sessionEditFormModel = {
+        ...data,
+        submissionEndDate: submissionStartDate,
+        submissionEndTime: submissionStartTime.hour > submissionEndTime.hour ? submissionStartTime : submissionEndTime,
+      };
+    } else if (startDate.toISOString() === endDate.toISOString() && submissionStartTime.hour > submissionEndTime.hour) {
```

Comment:
You're missing the case where:
- startDate is same as endDate
- startTime.hour is same as endTime.hour
- startTime.minute is greater than endTime.minute

This case only occurs when startTime is 23:59 and endTime is 23:00. Here's a quick test:
- Set submission opening time to 5 Jul 2024, 23:59
- Set submission closing time to 6 Jul 2024, 23:00
- Change submission opening date to 6 Jul 2024
- The submission closing time should change to 23:59, but it remains as 23:00 (see screenshot)
<img width="1304" alt="Screenshot 2023-07-11 at 6 52 07 PM" src="https://github.com/TEAMMATES/teammates/assets/48304907/ddc07b6f-f753-418a-be38-58c8c565de3a">

Reply:
Thanks for the catch! I have fixed it :)